rust/lance/src/io/commit/conflict_resolver.rs

```diff
@@ -827,13 +1002,152 @@ impl<'a> TransactionRebase<'a> {
+                        other_version,
+                    )?;
+                    Ok(())
+                }
+                Operation::Update {
+                    mem_wal_to_flush, ..
+                } => {
+                    if mem_wal_to_flush.is_some() {
+                        Ok(())
+                    } else {
+                        Err(self.incompatible_conflict_err(
+                            other_transaction,
+                            other_version,
+                            location!(),
+                        ))
+                    }
+                }
+                Operation::UpdateConfig { .. }
+                | Operation::Rewrite { .. }
+                | Operation::CreateIndex { .. }
+                | Operation::ReserveFragments { .. } => Ok(()),
+                Operation::Append { .. }
+                | Operation::Overwrite { .. }
+                | Operation::Delete { .. }
+                | Operation::DataReplacement { .. }
+                | Operation::Merge { .. }
+                | Operation::Restore { .. }
+                | Operation::Project { .. } => Err(self.incompatible_conflict_err(
+                    other_transaction,
+                    other_version,
+                    location!(),
+                )),
+            }
+        } else {
+            Err(wrong_operation_err(&self.transaction.operation))
+        }
+    }
+
+    fn check_update_mem_wal_state_not_modify_same_mem_wal(
+        &self,
+        committed: &[MemWal],
+        to_commit: &[MemWal],
+        other_transaction: &Transaction,
+        other_version: u64,
+    ) -> Result<()> {
+        if !committed.is_empty() {
+            if committed.is_empty() || to_commit.is_empty() {
+                return Ok(());
+            }
+
+            if committed.len() > 1 {
+                return Err(Error::Internal {
+                    message: format!(
+                        "Committing MemWALs is not supported, but found committed: {:?}",
+                        committed
+                    ),
+                    location: location!(),
+                });
+            }
+
+            if to_commit.len() > 1 {
+                return Err(Error::NotSupported {
+                    source: format!(
+                        "Committing MemWALs is not supported, but found attempt to commit: {:?}",
+                        to_commit
+                    )
+                    .into(),
+                    location: location!(),
+                });
+            }
+
+            if let (Some(committed_mem_wal), Some(to_commit_mem_wal)) =
+                (committed.first(), to_commit.first())
+            {
+                if committed_mem_wal.id == to_commit_mem_wal.id {
+                    return Err(self.incompatible_conflict_err(
+                        other_transaction,
+                        other_version,
+                        location!(),
+                    ));
+                }
+            }
```

Comment:
Maybe just use unwrap here?

```
let committed_mem_wal = committed.first().unwrap();
let to_commit_mem_wal = to_commit.first().unwrap();
```

This way if the logic above is refactored then we won't accidentally fall into a silent else.